Tiltfile

```diff
@@ -396,6 +401,77 @@ def prepare_all():
     )
     local(cmd, env = settings.get("kustomize_substitutions", {}))
 
+# create flavor resources from cluster-template files in the templates directory
+def flavors():
+    substitutions = settings.get("kustomize_substitutions", {})
+
+    template_list = [item for item in listdir("./test/infrastructure/docker/templates")]
+    template_list = [template for template in template_list if os.path.basename(template).endswith("yaml")]
+
+    for template in template_list:
+        deploy_worker_templates(template, substitutions)
+
+    local_resource(
+        name = "delete-all-workload-clusters",
+        cmd = kubectl_cmd + " delete clusters --all --wait=false",
+        auto_init = False,
+        trigger_mode = TRIGGER_MODE_MANUAL,
+        labels = ["flavors"],
+    )
+
+def deploy_worker_templates(template, substitutions):
+    # validate template exists
+    if not os.path.exists(template):
+        fail(template + " not found")
+
+    yaml = str(read_file(template))
+    flavor = os.path.basename(template).replace("cluster-template-", "").replace(".yaml", "")
+
+    # for the base cluster-template, flavor is "default"
+    flavor = os.path.basename(flavor).replace("cluster-template", "default")
+
+    # ssh replacements
+    for substitution in substitutions:
+        value = substitutions[substitution]
+        yaml = yaml.replace("${" + substitution + "}", value)
+
+    # if metadata defined for worker-templates in tilt_settings
+    if "worker-templates" in settings:
+        # first priority replacements defined per template
+        if "flavors" in settings.get("worker-templates", {}):
+            substitutions = settings.get("worker-templates").get("flavors").get(flavor, {})
+            for substitution in substitutions:
+                value = substitutions[substitution]
+                yaml = yaml.replace("${" + substitution + "}", value)
+
+        # second priority replacements defined common to templates
+        if "metadata" in settings.get("worker-templates", {}):
+            substitutions = settings.get("worker-templates").get("metadata", {})
+            for substitution in substitutions:
+                value = substitutions[substitution]
+                yaml = yaml.replace("${" + substitution + "}", value)
+
+    # programmatically define any remaining vars
+    substitutions = {
+        "NAMESPACE": settings.get("namespace"),
+        "KUBERNETES_VERSION": settings.get("kubernetes_version"),
+        "CONTROL_PLANE_MACHINE_COUNT": "1",
+        "WORKER_MACHINE_COUNT": "2",
+    }
```

Comment:
I really would like to make this simpler for the developers.
Can we start with just kustomize_substitutions in the first iteration (instead of introducing multiple layers of configurations)?

Reply:
Yep I think we can trim this stuff down, I think some of the substitutions were specific to CAPZ so we should be able to simply this. From a UX standpoint, do you think we should provide defaults for any of the substitutions in the existing templates like `WORKER_MACHINE_COUNT`, or should we rely on the user to explicitly state all substituted variables in `kustomize_substitutions`? If we're reading in custom templates, it could be a bit confusing for users to figure out which variables have default values and which ones don't.

Reply:
Agree, the simpler the better.
In the first iteration, all variables should be provided by the user either in substitutions or as envvariables